### PR TITLE
Rewrite: rederive an if code capability

### PR DESCRIPTION
**Pull request by kwitaszczyk**
An if code block capability must be rederived from a code array in case it had to be reallocated by the nginx configuration parser as it was already full.

### CODE REVIEW

File: src/http/modules/ngx_http_rewrite_module.c [hunk 1]
```diff
@@ -629,7 +631,7 @@ ngx_http_rewrite_if(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
 
     if (elts != lcf->codes->elts) {
         if_code = (ngx_http_script_if_code_t *)
```
Comment (bsdjhb) [inline src/http/modules/ngx_http_rewrite_module.c:631]: You could do the assignment and declaration here to make the diff smaller?

Comment (kwitaszczyk) [reply src/http/modules/ngx_http_rewrite_module.c:631]: Do you mean the assignment and declaration of `if_code_offset`? `lcf->codes->elts`, or a value depending on it (`if_code_offset` in this case), must be stashed before `ngx_conf_parse()` is called.

Comment (kwitaszczyk) [reply src/http/modules/ngx_http_rewrite_module.c:631]: Oh, I see what you mean. We can use `elts` to compute `if_code_offset` instead.

Comment (kwitaszczyk) [reply src/http/modules/ngx_http_rewrite_module.c:631]: This should look better. Thanks!

**Review by bsdjhb — APPROVED** [review]

**Review by qwattash — APPROVED** [review]

**Pull request merged**